### DISCUSSION TITLE
对于部分数据定义的疑问

### DISCUSSION THREAD

**Discussion by mxi-box** [ZsgsDesign/NOJ#796] [Q&A] [answered]
在尝试将NOJ转移到PostgreSQL上时发现了这个view里面的一段赋值。
public值为1是代表公开，2为非公开，但是migrant表**group**时定义public列类型为boolean，这个是否有一些冲突？

upd:在比赛的可见性上也用了boolean，但是却定义了012三个数字，可能有数据库类型错误。

https://github.com/ZsgsDesign/NOJ/blob/9e774bcb7b0892a4c4c5f53ec5a4840217e19dba/resources/views/group/create.blade.php#L184

**Comment by ZsgsDesign** [comment, marked as answer]
Public的1与2是一个BUG，应该使用1与0表示是、否，而不是1与2，**将会在`v1.0.0`修复**。

他们并没有被映射为Eloquent模型的boolean类型，因此在MySQL、Oracle与MSSQL使用上是没有问题的，PostgreSQL由于支持Boolean，会产生问题，可见性也是同理。这是由于NOJ在一开始使用SQL进行数据库版本管理，迁移时使用了一个SQL转migration工具，该工具将tinyint（定义为1字节0~255数据范围）自动转换成了boolean，而系统的本意是使用1字节整形而非布尔型，因此造成了问题。

**对于boolean的问题将会补丁修复为tinyinteger。**
